### PR TITLE
Few fixes and small changes.

### PR DESCRIPTION
**Pull request by Vizit0r**
1) fix of OPENARRAYOFU16 and OPENARRAYOFS16 declarations
2) DynArray "out of range" errors extended, now included info about curr index and array length
3) uPSUtils - takilng ansi string length from 0 index - changed to correct Length() calling.
4) added names forming for dynarrays when compiling.

### CODE REVIEW

File: Source/uPSUtils.pas [hunk 1]
```diff
@@ -318,10 +318,14 @@ function MakeHash(const s:  TbtString): Longint;
 {$IFDEF FPC}
   IPointer = PtrUInt;
 {$ELSE}
-  {$IFDEF CPUX64}
-  IPointer = IntPtr;
+  {$IFDEF VER140UP}
+    IPointer = NativeInt;
```
Comment (martijnlaan) [inline Source/uPSUtils.pas]: @Vizit0r What's the reason for this change? It seems unrelated to the list of changes listed in the pull commit. This change causes a whole load of 'W1024 Combining signed and unsigned types' warnings.
Fyi @carlokok

File: Source/uPSRuntime.pas [hunk 2]
```diff
@@ -4505,7 +4514,7 @@ function TPSExec.SetVariantValue(dest, Src: Pointer; desttype, srctype: TPSTypeR
       btPChar: pansichar(dest^) := pansichar(PSGetAnsiString(Src, srctype));
       btString:
         tbtstring(dest^) := PSGetAnsiString(Src, srctype);
-      btChar: tbtchar(dest^) := tbtchar(PSGetUInt(Src, srctype));
+      btChar: tbtchar(dest^) := PSGetAnsiChar(Src, srctype);
```
Comment (martijnlaan) [inline Source/uPSRuntime.pas]: @Vizit0r This change is not explained by the commit comment? Why was it made?

File: Source/uPSRuntime.pas [hunk 3]
```diff
@@ -4567,6 +4576,9 @@ function TPSExec.SetVariantValue(dest, Src: Pointer; desttype, srctype: TPSTypeR
           if srctype.BaseType = btClass then
             TObject(Dest^) := TObject(Src^)
           else
+          if srctype.BaseType = btVariant then
```
Comment (martijnlaan) [inline Source/uPSRuntime.pas]: @Vizit0r This change is not explained by the commit comment? Why was it made?

File: Source/uPSRuntime.pas [hunk 4]
```diff
@@ -10290,7 +10310,7 @@ function ClassCallProcConstructor(Caller: TPSExec; p: TPSExternalProcRec; Global
     v := NewPPSVariantIFC(Stack[CurrStack + 1], True);
   end else v := nil;
   try
-    Result := Caller.InnerfuseCall(FSelf, p.Ext1, {$IFDEF FPC}TPSCallingConvention(Integer(cc) or 64){$ELSE}cc{$ENDIF}, MyList, v);
+    Result := Caller.InnerfuseCall(FSelf, p.Ext1, TPSCallingConvention(Integer(cc) or 64), MyList, v);
```
Comment (martijnlaan) [inline Source/uPSRuntime.pas]: @Vizit0r This change is not explained by the commit comment? Why was it made?

Comment (Vizit0r) [reply Source/uPSRuntime.pas]: this change is need to explain for InnerfuseCall x86 that constructor called.
For InvokeCall.inc this has no matter.

And yes, it should be covered by conditional define USEINVOKECALL

File: Source/uPSRuntime.pas (hunk 3 above)
Comment (Vizit0r) [reply Source/uPSRuntime.pas]: canot say exactly, but i remember, that long time ago i have some bug with classes, thats why this line has been added. But that was really LONG time ago, so no details.
Are there any problems due to this fix observed?
